tests/snapshots/compile_file__self_ref.hvm.snap

```diff
@@ -2,4 +2,4 @@
 source: tests/golden_tests.rs
 input_file: tests/golden_tests/compile_file/self_ref.hvm
 ---
-Definition main is a reference to itself
```

Comment:
Can you make this still tests a self reference and have a separate test for main references?